ert/data/record/_record.py

```diff
@@ -496,8 +503,9 @@ async def load_collection_from_file(
             length=length,
             collection_type=RecordCollectionType.UNIFORM,
         )
-    return RecordCollection(
-        records=await ert.data.FileRecordTransformation().transform_output_sequence(
-            mime, file_path
+    else:
```

Comment:
Is this triggering a pylint warning ala uneccessary else after return?

Reply:
This needs to be fine-tuned since `xlsx` gets `mime-type=application/json` :shrug:
When using it directly in python:
```
In [2]: mimetypes.guess_type("resources/designmatrix.xlsx")
Out[2]: ('application/vnd.openxmlformats-officedocument.spreadsheetml.sheet', None)
```
So it sort-of belongs to the else branch.

Reply:
The issue is that we still depend on having a file serializer (which currently support only json and yaml) :
```python
guess = mimetypes.guess_type(str(values.get(basis, "")))[0]
    if guess:
        if not ert.serialization.has_serializer(guess):
            return DEFAULT_RECORD_MIME_TYPE
```
Are we to remove serializers and replacing them with transformation?

Reply:
The answer here is we're keeping the serializers, but they've been demoted to an implementation detail in the serializationtransformation…